elastic/kibana
# [Stateful sidenav] Breadcrumb switcher

In this PR I've added the solution nav switcher in the header for the new side navigation.
The switcher allows the user to switch between the different solution navigations.

Note: This PR **only** adds the switcher, the actual solution navigations IA will be added in a separate PR.

## Screenshot

<img width="399" alt="Screenshot 2024-03-13 at 12 36 12" src="https://github.com/elastic/kibana/assets/2854616/7c074461-4252-4cd4-8630-6e5c67a9800f">


## APIs exposed by the `navigation` plugin

### `addSolutionNavigation()`

This API is used to add a solution navigation to the header switcher.

```ts
// Example
plugin.navigation.addSolutionNavigation({
  id: 'security', // unique id
  title: 'Security', // Title to render in the switcher
  icon: 'logoSecurity', // Icon to render in the switcher
  homePage: 'dashboards', // The page to navigate to when the user clicks on the solution navigation
  navigationTree$: of({
    body: [
      // This is the exact same navigation tree that is used in serverless
      {
        type: 'navGroup',
        id: 'security_project_nav',
        title: 'Security',
        icon: 'logoSecurity',
        breadcrumbStatus: 'hidden',
        defaultIsCollapsed: false,
        children: [
          { link: 'dashboards'},
          ...
        ],
      },
    ],
  }),
});
```

### `isSolutionNavigationEnabled()`

This API is used to check if the solution navigation is enabled.

```ts
// Example
const isEnabled = plugin.navigation.isSolutionNavigationEnabled();

if (isEnabled) {
  // The feature flag is enabled...
}
```

## How to test

1. Add the following `kibana.yml` config:

```yml
navigation.solutionNavigation.featureOn: true
navigation.solutionNavigation.enabled: true
navigation.solutionNavigation.optInStatus: visible
navigation.solutionNavigation.defaultSolution: es
```

2. Launch Kibana
3. Verify that the solution navigation switcher is visible in the header

Fixes https://github.com/elastic/kibana/issues/177921
